### PR TITLE
Enable fmt

### PR DESCRIPTION
**Pull request by xMAC94x**
## Checklist

* [ ] I've added tests for all code changes and additions (where applicable)
* [ ] I've added a demonstration of the new feature to one or more examples
* [ ] I've updated the book to reflect my changes
* [ ] Usage of new public items is shown in the API docs

### CODE REVIEW

File: .rustfmt.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 hard_tabs = false
-merge_imports = true
```
Comment (AngelOnFira) [inline .rustfmt.toml:2]: Is this an equivalent swap?

**Review by AngelOnFira — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -49,8 +49,8 @@ jobs:
         if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
 
       # run rustfmt and clippy checks, but only once
-      #- run: cargo +nightly fmt --all -- --check
-      #  if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
+      - run: cargo +nightly fmt --all -- --check
+        if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
 
       #- run: cargo +nightly clippy -Z unstable-options --workspace --all-targets --all-features
       #  if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
```
Comment (AngelOnFira) [inline .github/workflows/ci.yml:56]: If you made fixes for Clippy, should this be enabled as well?

File: .rustfmt.toml (hunk 1 above)
Comment (xMAC94x) [reply .rustfmt.toml:2]: yep

File: .github/workflows/ci.yml (hunk 2 above)
Comment (xMAC94x) [reply .github/workflows/ci.yml:56]: i made some easy clippy fixes, but there are further ones i couldnt fix

**Pull request merged**